src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges } from '@angular/core'
+import { Segment } from '../../../core/models/segment'
+import { ConfigurationService } from '../../../shared/services/configuration-service'
+import { StudioConfiguration } from '../../../shared/services/studio-configuration'
+import { Tv2Action, Tv2VideoClipAction } from '../../../shared/models/tv2-action'
+import { Subscription } from 'rxjs'
+import { MediaDataService } from '../../../shared/services/media-data.service'
+import { MediaData } from '../../../shared/services/media-data'
+
+@Component({
+  selector: 'sofie-mini-shelf',
+  styleUrls: ['./mini-shelf.component.scss'],
+  templateUrl: './mini-shelf.component.html',
+})
+export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() public segment: Segment
+  @Input() public videoClipAction: Tv2VideoClipAction | undefined
+
+  @Output() public onClickExecuteActionEmitter: EventEmitter<Tv2Action> = new EventEmitter()
+
+  protected mediaDuration: number = 0
+  private configurationMediaPreviewUrl: string
+  private mediaDataSubscription: Subscription
+  private configurationServiceSubscription: Subscription
+
+  constructor(
+    private readonly configurationService: ConfigurationService,
+    private readonly mediaDataService: MediaDataService
+  ) {}
+
+  public ngOnInit(): void {
+    this.configurationServiceSubscription = this.configurationService.getStudioConfiguration().subscribe((configuration: StudioConfiguration) => {
+      this.configurationMediaPreviewUrl = configuration.data.settings.mediaPreviewUrl
+    })
+    this.updateMediaDataDuration()
+  }
+
+  private updateMediaDataDuration(): void {
+    if (this.segment.metadata?.miniShelfVideoClipFile !== undefined) {
+      this.mediaDataSubscription = this.mediaDataService.getMediaDurationById(this.segment.metadata.miniShelfVideoClipFile).subscribe((mediaData: MediaData) => {
+        this.mediaDuration = mediaData.duration
+      })
+    }
+  }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    if ('segment' in changes) {
+      this.updateMediaDataDuration()
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.mediaDataSubscription?.unsubscribe()
+    this.configurationServiceSubscription?.unsubscribe()
+  }
+
+  protected get mediaPreviewUrl(): string {
+    const url: string = `${this.configurationMediaPreviewUrl}/media/thumbnail/${this.segment.metadata?.miniShelfVideoClipFile}`
+    return this.configurationMediaPreviewUrl ? url : 'assets/sofie-logo.svg'
```

Comment:
In such case then I'll need an asset image to use instead. Also in case thumbnail image is not reachable (fails to load or is non-existent ) and in:
`  protected handleMissingImage(event: Event): void {
    ;(event.target as HTMLImageElement).src = 'assets/sofie-logo.svg'
  }`

Reply:
Yes, we should ask Jacob what he wants in that case